File: ECommerceFront/src/app/models/document.ts
import { DocumentStatus } from './documentStatus';
import { DocumentDetail } from './documentDetail';
import { DocumentType } from './documentType';
import { Customer } from './customer';

export class Document {
    documentId: number;
    code: string;
    date: Date;
    sum: number;
    documentStatus: DocumentStatus;
    documentType: DocumentType;
    documentDetails: Array<DocumentDetail>;
    customer: Customer;

    constructor() {
        this.documentId = null;
        this.code = '';
        this.date = new Date();
        this.sum = 0;
        this.documentStatus = new DocumentStatus();
        this.documentType = new DocumentType();
        this.documentDetails = new Array<DocumentDetail>();
        this.customer = new Customer();
    }

    assignObject(init?: Partial<Document>) {
        Object.assign(this.customer, init.customer);
        Object.assign(this.documentStatus, init.documentStatus);
        Object.assign(this.documentType, init.documentType);
        Object.assign(this.documentDetails, init.documentDetails);
        Object.assign(this, init);
        // Object.assign(this.documentDetails, init.documentDetails);
    }

    // assignObject(document: Document) {
    //     t
    // }
}

File: ECommerceAPI/ECommerceAPI/Models/Document.cs
using ECommerceAPI.Base.Attributes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ECommerceAPI.Models
{
    public class Document
    {
        public int DocumentId { get; set; }
        [ChildValidation(new string[] { "Code;documenttype_required" })]
        public DocumentType DocumentType { get; set; }
        [ChildValidation(new string[] { "Code;documentstatus_required" })]
        public DocumentStatus DocumentStatus { get; set; }
        public string Code { get; set; }
        [Required(ErrorMessage = "date_required")]
        public DateTime Date { get; set; }
        public decimal Sum { get; set; }
        [ChildValidation(new string[] { "CustomerId;customer_required;DocumentType.Code=bill" })]
        public Customer Customer { get; set; }
        public List<DocumentDetail> DocumentDetails { get; set; }
        public int CustomerId { get; set; }
        public int DocumentStatusId { get; set; }
        public int DocumentTypeId { get; set; }

        public Document()
        {
            DocumentDetails = new List<DocumentDetail>();
        }
    }
}
